src/EditorFeatures/CSharpTest/UseConditionalExpression/UseConditionalExpressionForReturnTests.cs

```diff
@@ -550,6 +550,56 @@ class C
     {
         return ref true ? ref i : ref j;
     }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseConditionalExpression)]
+        public async Task TestOnYieldReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    int M()
+    {
+        [||]if (true)
+        {
+            yield return 0;
+        }
+        else
+        {
+            yield return 1;
+        }
+    }
+}",
+@"
+class C
+{
+    int M()
+    {
+        yield return true ? 0 : 1;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseConditionalExpression)]
+        public async Task TestNotOnMixedYields()
```

Comment:
:bulb: Can you add a test for `if` without an explicit `else`? (as shown in the bug report)?

Reply:
Definitely!